projects/core/src/site-context/site-context.module.ts

```diff
@@ -1,16 +1,40 @@
 import { ModuleWithProviders, NgModule } from '@angular/core';
+import {
+  ConfigInitializer,
+  CONFIG_INITIALIZER,
+} from '../config/config-initializer/config-initializer';
 import { provideDefaultConfigFactory } from '../config/config-providers';
 import { provideConfigValidator } from '../config/config-validator/config-validator';
 import { StateModule } from '../state/index';
 import { baseSiteConfigValidator } from './config/base-site-config-validator';
+import { SiteContextConfigLoaderService } from './config/config-loader/site-context-config-loader.service';
 import { defaultSiteContextConfigFactory } from './config/default-site-context-config';
+import { SiteContextConfig } from './config/site-context-config';
 import { SiteContextEventModule } from './events/site-context-event.module';
+import { BASE_SITE_CONTEXT_ID } from './providers/context-ids';
 import { contextServiceMapProvider } from './providers/context-service-map';
 import { contextServiceProviders } from './providers/context-service-providers';
 import { siteContextParamsProviders } from './providers/site-context-params-providers';
 import { SiteContextStoreModule } from './store/site-context-store.module';
 
-// @dynamic
+/**
+ * Initializes the site context config
+ */
+export function initSiteContextConfig(
+  configLoader: SiteContextConfigLoaderService,
+  config: SiteContextConfig
+): ConfigInitializer | null {
+  /**
+   * Load config for `context` from backend only when there is no static config for `context.baseSite`
+   */
+  if (!config.context || !config.context[BASE_SITE_CONTEXT_ID]) {
+    return {
+      scopes: ['context'],
+      configFactory: () => configLoader.loadConfig(),
+    };
+  }
```

Comment:
This new logic should be executed for >= 3.2
```ts
if(featureConfigService.isLevel('3.2')) {
  ...
}
```

Otherwise if customer stays in '3.1' feature level in his config, then both old and new logic would be executed.

Reply:
Done